src/managed_data/_api.py

```diff
@@ -2,15 +2,13 @@
 with the data API. functions here simply wrap API endpoints."""
 
 import os
-from typing import Literal, Optional, Union
+from typing import Optional, Union
 
 import requests
 from beartype import beartype
 from deeporigin.exceptions import DeepOriginException
-from deeporigin.managed_data.client import DeepOriginClient
-
-# types of rows
-RowType = Literal["row", "workspace", "database"]
+from deeporigin.managed_data.client import Client, DeepOriginClient
+from deeporigin.managed_data.schema import RowType
 
 
 def _get_default_client(client):
```

Comment:
For thought: why separate the client class and the methods below. Why not integrate them together so the methods and data are encapsulated together into one class? In this design, `_get_default_client` wouldn't be needed, or at least would take a different form.

Reply:
2 reasons:

- historically i started out writing everything as functions. 
- kenny then convinced me (rightly so) that i should adopt this structure so that i could swap out clients making tests easier and more transparent. 

i like the pure functional approach that exists here, and i think it requires less boilerplate. were i to do what you suggest (if i understand you correctly), then there is one class that specified the client and all the methods in `_api`. that's exactly what i wanted to avoid -- i wanted a separation of the client from the low-level functions. were i to do your suggestion, i'd have to re-implement or overload those classes in the tests. 

finally, i agree that `_get_default_client` is clunky, but that's hard to avoid given how python initializes function default arguments. i would have liked to write this:


```python
def some_low_level_func(arg, client=DeepOriginClient):
    pass 
```

but that would require re-imports to hot-swap clients (or change config variables)